x/evidence/keeper/infraction.go

```diff
@@ -40,6 +40,13 @@ func (k Keeper) handleEquivocationEvidence(ctx context.Context, evidence *types.
 	}
 
 	if len(validator.GetOperator()) != 0 {
+		// get the consAddr again, this is because validator might've rotated it's key.
+		valConsAddr, err := validator.GetConsAddr()
+		if err != nil {
+			panic(err)
+		}
+		consAddr = valConsAddr
+
 		if _, err := k.slashingKeeper.GetPubkey(ctx, consAddr.Bytes()); err != nil {
 			// Ignore evidence that cannot be handled.
 			//
```

Comment:
I agree with the rabbit, just return an error and the app will panic anyway because this error will be returned from BeginBlock and returning errors from there cause a panic.

Reply:
++ return an error (generally speaking we shouldn't panic in APIs that return an error)